doc/rewriting-services.md

```diff
@@ -89,10 +89,7 @@ still pass.
 the route using a `pattern`. A `pattern` (e.g. `pattern: ':param1/:param2'`) is
 the simplest way to declare the route, also the most readable, and will be
 useful for displaying a badge builder with fields in the front end and
-generating badge URLs programmatically. One limitation to keep in mind is that,
-at present, the trailing parameter of a pattern can't be optional. If the last
-part of a route is optional, like a branch, you will need to use a `format`
-regex string (e.g. `format: '([^/]+/[^/]+)'`).
+generating badge URLs programmatically.
```

Comment:
Do we have any guidance/tips on using `pattern` anywhere else? I've learned about various shortcuts/patterns that path-to-rexep` supports from places like comments folks have left on PRs that I didn't know about. 

If not maybe we could link out to https://github.com/pillarjs/path-to-regexp?

Reply:
I think that link is in the tutorial. What are the tips you'd give? Maybe we can start creating a new section.

Reply:
One that comes to mind: you can't use non-capturing groups in a `pattern`.

Reply:
That's a good example of one. I'd tried to fight it to make non-capturing groups work (obviously unsuccessfully 😆 ), but I wasn't sure if it wouldn't work because it wasn't possible or if I just didn't have the syntax/escapes right. 

The other thing I just learned the other day are some of the shorthands like `*` and `+` can be added directly to the param (I think one example is `:foo*` works as the shorthand for `:foo(.*)` .

Reply:
I'm still thinking that some pattern tips _may_ be useful somewhere, but doesn't need to be here nor part of this PR IMO